soft4pes/control/lin/state_space_curr_ctr.py

```diff
@@ -0,0 +1,254 @@
+        self.uc_km1 = np.zeros(2)
+        self.i_ref_seq_dq = i_ref_seq_dq
+
+        self.sim_data = {
+            'ig_ref': [],
+            'u': [],
+            't': [],
+        }
+
+    def __call__(self, sys, conv, t):
+        """
+        Perform control.
+
+        Parameters
+        ----------
+        sys : system object
+            System model.
+        conv : converter object
+            Converter model.
+        t : float
+            Current time [s].
+
+        Returns
+        -------
+        1 x 3 ndarray of floats
+            Modulating signal.
+        """
+
+        # Get the grid voltage
+        vg = sys.get_grid_voltage(t)
+
+        # Get the reference for current step
+        ic_ref_dq = self.i_ref_seq_dq(t)
+
+        # Calculate the transformation angle
+        theta = np.arctan2(vg[1], vg[0])
+
+        # Get the current in dq frame
+        ic_dq = alpha_beta_2_dq(sys.x, theta)
+
+        # Maximum voltage
+        u_max = conv.v_dc / 2
+
+        #Consider the filter capacitor voltage equals to the grid voltage (In case: Without considering the filter)
+        uf_dq = vg
+
+        # Compute the converter voltage reference in dq frame using the state space controller with anti-windup
+        uc_dq = self.state_space_controller(ic_dq, ic_ref_dq, uf_dq, u_max)
+
+        # Transform the converter voltage reference back to abc frame
+        uc_abc = dq_2_abc(uc_dq, theta)
+
+        u_k = uc_abc / (conv.v_dc / 2)
+
+        # Save controller data
+        ig_ref = dq_2_alpha_beta(ic_ref_dq, theta)
+        self.save_data(ig_ref, u_k, t)
+
+        # Ensure modulating signal within -1 and 1
+        return np.clip(u_k, -1, 1)
+
+    def get_state_space_ctr_pars(self):
+        """
+        Calculate state-space controller parameters.
+        
+        Returns
+        -------
+        SimpleNamespace
+            Controller parameters.
+        """
+        Ts_pu = self.Ts_pu
+        Rf = self.Rf
+        Xf = self.Xf
+
+        # Closed-loop controller bandwidth (10x crossover frequency)
+        alpha_c = 2 * np.pi / 10 / Ts_pu
+
+        # Consider delta equals to one due to not considering delay
+        delta = 1
+
+        # Coefficients
+        phi = np.exp((-Rf / Xf) * Ts_pu) * delta
+        landa = (delta - phi) / Rf
+
+        # The closed-loop pole locations
+        p_1 = 0
+        p_2 = np.exp(-alpha_c * Ts_pu)
+        p_3 = p_2
+
+        # Coefficients
+        k_2 = -p_1 - p_2 - p_3 + phi + 1
+        k_1 = (p_1 * p_2 + p_1 * p_3 + p_2 * p_3 + k_2 * phi + k_2 -
+               phi) / landa
+
+        # State feedback gain
+        K_i = np.array([k_1, 0, k_2])
+
+        # Integral gain
+        k_ii = (-p_1 * p_2 * p_3 + k_1 * landa - k_2 * phi) / landa
+
+        # Feedforward gain
+        k_ti = k_ii / (1 - p_3)
+
+        return SimpleNamespace(delta=delta, K_i=K_i, k_ii=k_ii, k_ti=k_ti)
+
+    def state_space_controller(self, ic_dq, ic_ref_dq, uf_dq, u_max):
```

Comment:
ic is also ig. If you use ic, please add a clear line comment somewhere in the code where you explain that they are the same